### PR TITLE
Fix `ky.extend()` not inheriting its parent and add `ky.create()`

### PR DESCRIPTION
**Pull request by szmarczak**

### CODE REVIEW

**Comment by szmarczak** [comment]
TODO: mutable defaults (#116 should fix that)

File: readme.md [hunk 1]
```diff
@@ -242,16 +242,16 @@ Throw a `HTTPError` for error responses (non-2xx status codes).
 
 Setting this to `false` may be useful if you are checking for resource availability and are expecting error responses.
 
-### ky.extend(defaultOptions)
+### ky.create(defaultOptions)
```
[the remark this answers is not in the document]
Comment (sholladay) [reply readme.md]: So `.create()` is basically like using `.extend()` but passing all options with `undefined` as the value in order to unset them. Is that right?

[the remark this answers is not in the document]
Comment (sindresorhus) [reply readme.md]: Correct. It's a lower-level building block. This idea is from Got: https://github.com/sindresorhus/got/blob/master/advanced-creation.md

[the remark this answers is not in the document]
Comment (szmarczak) [reply readme.md]: fixed

**Pull request merged**